pkg/util/xds/v3/watchdog_callbacks.go

```diff
@@ -35,6 +35,30 @@ type watchdogStreamState struct {
 
 var _ envoy_xds.Callbacks = &watchdogCallbacks{}
 
+// RestStreamID is used in the non-streaming REST context
+const RestStreamID = int64(-1)
+
+func (cb *watchdogCallbacks) hasStream(streamID int64) bool {
+	cb.mu.RLock() // read access to the map of all ADS streams
+	defer cb.mu.RUnlock()
+	_, ok := cb.streams[streamID]
+	return ok
+}
+
+func (cb *watchdogCallbacks) OnFetchRequest(ctx context.Context, req *envoy_discovery.DiscoveryRequest) error {
+	// Open up a new "stream" state, which all REST requests use, if
+	if cb.hasStream(RestStreamID) {
+		return nil
+	}
+
+	if err := cb.OnStreamOpen(ctx, RestStreamID, req.TypeUrl); err != nil {
+		return err
+	}
+	// TODO: could also register a TTL on the REST stream to clean it up if there is no activity over a certain period,
+	// 		 since it will currently never be closed once opened
```

Comment:
If you think this is necessary for right now, happy to implement it, though I think we're safe leaving this as a follow up because:
* max 1 background process
* requires at least 1 request to start